rust-runtime/smithy-xml/src/encode.rs

```diff
@@ -60,42 +60,55 @@ impl<'a> XmlWriter<'a> {
     pub fn start_el<'b, 'c>(&'c mut self, tag: &'b str) -> ElWriter<'c, 'b> {
         write!(self.doc, "<{}", tag).unwrap();
         ElWriter {
-            doc: self.doc,
+            doc: Some(self.doc),
             start: tag,
         }
     }
 }
 
 pub struct ElWriter<'a, 'b> {
     start: &'b str,
-    doc: &'a mut String,
+    doc: Option<&'a mut String>,
 }
 
 impl<'a, 'b> ElWriter<'a, 'b> {
     pub fn write_attribute(&mut self, key: &str, value: &str) -> &mut Self {
-        write!(self.doc, " {}=\"{}\"", key, escape(value)).unwrap();
```

Comment:
can you add a private function along the lines of below that has a comment explaining why this is an option, why it won't be `None`, and generally, what's going on?
```rust
fn doc(&mut self) -> &'a mut String {
  // ...
}
```